src/ts-node.ts

```diff
@@ -38,12 +38,24 @@ function loadTSConfig(root: string): TSConfig | undefined {
       tsconfigPath,
       fs.readFileSync(tsconfigPath, 'utf8'),
     ).config
-    if (!tsconfig || !tsconfig.compilerOptions) {
+    const tsconfigParsed = typescript.parseJsonConfigFileContent(
+      tsconfig,
+      typescript.sys,
+      root,
+    )
+    if (!tsconfigParsed || !tsconfigParsed.options) {
       throw new Error(
         `Could not read and parse tsconfig.json at ${tsconfigPath}, or it ` +
         'did not contain a "compilerOptions" section.')
     }
-    return tsconfig
+    return {
+      compilerOptions: {
+        ...tsconfigParsed.options,
+        target: tsconfigParsed.options.target === undefined ?
+          tsconfigParsed.options.target :
+          typescript.ScriptTarget[tsconfigParsed.options.target],
```

Comment:
This bit is because `parseJsonConfigFileContent()` transforms "target" into its numerical enum value, so we have to change it back to a string to use with `tsnode.register()`.